test/Table_Tests/src/IO/Fetch_Spec.enso

```diff
@@ -517,6 +520,27 @@ add_specs suite_builder =
             Test_Environment.unsafe_with_environment_override "ENSO_LIB_HTTP_CACHE_MAX_TOTAL_CACHE_LIMIT" "101%" <|
                 LRUCache.new . getSettings . getTotalCacheLimit . should_equal (TotalCacheLimit.Percentage.new 0.2)
 
+        group_builder.specify "Cache should be cleared when a reload is detected" <|
+            HTTP.fetch base_url_with_slash+'test_download?length=10'
+            HTTP.fetch base_url_with_slash+'test_download?length=11'
+            HTTP.fetch base_url_with_slash+'test_download?length=12'
+            get_num_response_cache_entries . should_equal 3
+
+            fake_reload
+
+            get_num_response_cache_entries . should_equal 3  # Cleaning is not triggered until the next request
+            HTTP.fetch base_url_with_slash+'test_download?length=13'
+            get_num_response_cache_entries . should_equal 1
+            HTTP.fetch base_url_with_slash+'test_download?length=14'
+            HTTP.fetch base_url_with_slash+'test_download?length=15'
```

Comment:
```suggestion
            HTTP.fetch base_url_with_slash+'test_download?length=10'
            get_num_response_cache_entries . should_equal 1
            HTTP.fetch base_url_with_slash+'test_download?length=14'
            HTTP.fetch base_url_with_slash+'test_download?length=15'
```

What if we fetch a thing that was cached before? IMO this could be useful to test. But not strictly necessary.

Reply:
Done